packages/cactus-plugin-keychain-azure-kv/src/test/typescript/integration/plugin-keychain-azure-kv.test.ts

```diff
@@ -0,0 +1,100 @@
+import test, { Test } from "tape-promise/tape";
+
+import { v4 as uuidv4 } from "uuid";
+
+import { LogLevelDesc } from "@hyperledger/cactus-common";
+
+import {
+  IPluginKeychainAzureKvOptions,
+  PluginKeychainAzureKv,
+} from "../../../main/typescript/public-api";
+
+import { AzureCredentialType } from "../../../main/typescript/plugin-keychain-azure-kv";
+
+const logLevel: LogLevelDesc = "TRACE";
+
+test("get,set,has,delete alters state as expected for AzureCredentialType.InMemory", async (t: Test) => {
+  const options: IPluginKeychainAzureKvOptions = {
+    instanceId: uuidv4(),
```

Comment:
@jagpreetsinghsasan FYI this is where you can use the mock class and pass it in. I think you'll also need a little extra logic in the constructor of the plugin to examine if a `SecretClient` was passed in or not. If it was, then use that, otherwise proceed to examine if the credentials were passed in so that we could create our own `SecretClient` etc...

If the if conditions get too complicated/messy I can recommend the provider pattern where you'd define a provider function accepting credentials and then you'd have a real provider that instantiates `SecretClient` and another would just ignore the passed in credentials and return the mock.
This would definitely be the cleaner solution compared to what I outlined above in my first paragraph.